listenbrainz_spark/utils.py

```diff
@@ -186,18 +186,23 @@ def delete_dir(path, recursive=False):
                  For non-empty directory set recursive to 'True'.
     """
     try:
-        hdfs_connection.client.delete(path, recursive=recursive)
+        deleted = hdfs_connection.client.delete(path, recursive=recursive)
+        if not deleted:
+            raise HDFSDirectoryNotDeletedException('', path)
+        return deleted
     except HdfsError as err:
         raise HDFSDirectoryNotDeletedException(str(err), path)
 
 def get_status(path):
```

Comment:
This is very nitty, but I'm not sure if `get_status` is the best name for this function. `directory_exists` is much easier to understand.

Reply:
path_exists is better IMO, can be a file too.